### ISSUE TITLE
Throttle not working, battery shows 0%.

### ISSUE THREAD

**Issue by MrKris7100** [A-Emile/Lime_Gen3_IoT_Replacement#15]
Hello i succesfully flaashed original controler with hacked firmware. I wired up ESP with the scotter and uploaded your code. Scooter is powering up then i can connect to it using your app normally. When i am clicking unlock button the scooter unlocks, shows 0km/hs. And now there is issue. Scooter's display shows 0% on battery. I measured battery with multimeter and it's fully charged. Second problem is scooter doesn't react for the throttle and because of this it's not driving. Strange thing is when i unlock scooter the breaking light blinks a few times then stops and start blinking again. When i lock the scooter or disconnect it lock's itself normally and breaking light stops blinking.

Any advice about this issue? This scooter version is 1.04 (IoT PCB). I tried it with hacked firmware and original IoT with soldered jumper cable to make it boot but it behaviors simillar. Diffrence is it showing battery percentage correctly, is all time unlocked, breaking light reacts to break handle. But still doesn't react for the throttle.

Maybe i did something wrong in the wiring because i disconnected controller and take it to the home i had no IoT plug and just plugged ESP directly to controller but there are two sets of cables coming from controller with same collors. One is with 5 wires, second one is with 6. Maybe i taked wrong cable to connect RX and TX to the ESP? Does it matter?

<img src="https://github.com/A-Emile/Lime_Gen3_IoT_Replacement/assets/14978405/3f7b405b-6de5-4440-b8e2-f52ed0617d9a" width="256">
<img src="https://github.com/A-Emile/Lime_Gen3_IoT_Replacement/assets/14978405/d8fea30c-6bd3-4f0f-a351-d7ea20144ff0" width="256">
 Some pictures for addition.

**Comment by A-Emile** [comment]
Have you flashed the arduino fw i provided here: https://github.com/A-Emile/Lime_Gen3_IoT_Replacement/releases/tag/latest ?

**Comment by MrKris7100** [comment]
Well, actually i flashed code just from master branch not from release. I have ESP32 Wroom i guess it doesn't have shock sensor so i should flash version without sleep? Those break light flashes are from triggered alarm?

**Comment by A-Emile** [comment]
Yep thats the Problem probably. Try flashing the one without sleep. It should Work..

**Comment by FrezzZv2** [comment]
Did you ever figure it out? I'm facing the same issue right now, except I'm already using the "without sleep" fw. One of my scooters works perfectly, but the second one shows 0% baterry 0kmh & every light is off + throttle does nothing.

**Comment by G-Redos** [comment]
Try flashing the silver controller next to the battery.
There are 2 versions, 20km and 25km files.

**Comment by gc9n** [comment]
> Try flashing the silver controller next to the battery. There are 2 versions, 20km and 25km files.

The lock
Or the unlock ones ?

**Comment by aIecxs** [comment]
The 3.3v HIGH Level is maybe not high enough for 5v HIGH level, so behavior is undefined. This may help to understand the problem.
[![Levels.png](https://next-hack.com/wp-content/uploads/2020/02/Levels.png)](https://next-hack.com/index.php/2020/02/15/how-to-interface-a-3-3v-output-to-a-5v-input)
[How to interface a 3.3V output to a 5V input](https://next-hack.com/index.php/2020/02/15/how-to-interface-a-3-3v-output-to-a-5v-input)
